### PR TITLE
docs: add new test section for CSV Injection (WSTG-INPV-21)

### PR DESCRIPTION
**Pull request by m4sh-wacker**
This PR addresses a gap in the Input Validation Testing section regarding spreadsheet/CSV export security (CSV Injection / Formula Injection).

## What did this PR accomplish?

- Added a new testing guide: `21-Testing_for_CSV_Injection.md` (**WSTG-INPV-21**).
- Updated the `07-Input_Validation_Testing/README.md` index to include the new section.
- Covered safe, reproducible validation steps using benign formulas (e.g., `=1+1`) plus an optional `HYPERLINK()` example with a safe placeholder domain, and included an Excel save/re-open regression check because mitigations can degrade after save/reopen.
- Documented impact considerations (deception/phishing-style workflows, output manipulation, potential data exposure), and noted that higher impact may be possible in some environments via legacy spreadsheet features (e.g., DDE) without embedding weaponizable command-execution payloads.

### CODE REVIEW

**Comment by Galaxy-sc** [comment]
Hi! I see the linter is failing. I have the fixed version ready on my end; would you like me to push the fixes or post them as a suggestion for you?

**Comment by m4sh-wacker** [comment]
Hi @Galaxy-sc 
Thanks! Feel free to go ahead and push the fixes.

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+- A safe HTTP listener (local or internal test host) to observe clicks/requests when validating `HYPERLINK()` behavior.
+- A text editor to inspect the raw CSV and confirm whether any cell begins with a formula-triggering prefix.
+
+## References
+- [OWASP CSV Injection](https://owasp.org/www-community/attacks/CSV_Injection)
+- [CWE-1236: Improper Neutralization of Formula Elements in a CSV File](https://cwe.mitre.org/data/definitions/1236.html)
+- [HackerOne Report #72785: CSV Injection](https://hackerone.com/reports/72785)
+- [HackerOne Report #118582: CSV Injection](https://hackerone.com/reports/118582)
+- [HackerOne Report #1748961: CSV Injection](https://hackerone.com/reports/1748961)
+- [HackerOne Report #244292: CSV Injection](https://hackerone.com/reports/244292)
```
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md:169]:
```suggestion
# Testing for CSV Injection

|ID|
|--|
|WSTG-INPV-21|

## Summary

CSV Injection (also known as Formula Injection) occurs when an application embeds untrusted, user-controlled input into CSV (or other spreadsheet-compatible) exports and the resulting file is opened in a spreadsheet program (e.g., Microsoft Excel, LibreOffice Calc). Spreadsheet applications may interpret certain cell values as formulas, which can lead to security issues such as user deception (phishing-style workflows), manipulation of spreadsheet output, or data exfiltration. In some environments, formula injection can be escalated to higher impact via spreadsheet “gadgets” and legacy features (e.g., DDE / Dynamic Data Exchange behaviors), potentially reaching command execution on the workstation that opens the file—typically dependent on client configuration and/or user interaction.

A key characteristic of this issue is that the vulnerability often manifests only when the exported file is opened by a user (e.g., an administrator, finance, or support) in a spreadsheet application.

### Formula-triggering prefixes

Cells beginning with the following characters may be interpreted as formulas by spreadsheet software:

- Equals (`=`)
- Plus (`+`)
- Minus (`-`)
- At (`@`)
- Tab (`0x09`)
- Carriage return (`0x0D`)
- Line feed (`0x0A`)
- Full-width (double-byte) variants such as `＝`, `＋`, `－`, `＠` (depending on locale/application behavior)

> Important (Excel behavior): Microsoft Excel may remove quotes or escape characters from CSV cells when a file is saved and re-opened. As a result, some commonly suggested mitigations can fail after save/reopen and previously escaped formulas may become active again.

Also note that it is not sufficient to ensure the *overall* untrusted input does not start with a dangerous character. Attackers may inject separators and quoting to start a new cell, placing the dangerous character at the beginning of a cell.

## Test Objectives

- Identify CSV/spreadsheet export features that include untrusted input.
- Verify whether attacker-controlled values are interpreted as formulas when the export is opened in common spreadsheet applications.
- Check whether separator/quote injection can move a dangerous prefix to the start of a cell.
- Validate whether mitigations remain effective in Microsoft Excel after saving and re-opening the CSV.
- Assess practical impact based on who opens the export and how it is used.

## How to Test

### 1. Identify CSV Export Functionality and Data Sources

Locate features that generate CSV/TSV or “export to spreadsheet” content:

- Reports (users, transactions, audit logs, tickets)
- Admin dashboards exporting lists
- Email attachments generated by the application
- Scheduled exports / integrations

Identify untrusted data sources that can end up in the export:

- User profiles (name, email, company)
- Free-text fields (comments, ticket subjects, notes)
- Imported/integrated external data (webhooks, CRM sync, partner feeds)

Document which roles can trigger the export and which roles are likely to open it.

### 2. Place Benign, Detectable Formula-Like Values into Candidate Fields

Use harmless payloads to detect formula evaluation (avoid payloads that execute commands or perform uncontrolled network access).
Test values that begin with each formula-triggering character:

- `=1+1`
- `+1+1`
- `-1+1`
- `@SUM(1,1)`
- `=HYPERLINK("http://example.invalid/leak?test=1", "Click Me")`

Notes:

- The `HYPERLINK()` case is useful to demonstrate realistic impact (deception/phishing-style flows, or potential metadata exposure when a link is clicked/opened). Use a controlled endpoint during testing (e.g., a local listener or an internal test host).
- Do not use external “real attacker” infrastructure in validation.

Also test control-character and Unicode variants (where input handling allows it):

- A value that begins with a tab character followed by `=1+1` (TAB + `=1+1`)
- Full-width prefix variants (e.g., `＝1+1`)

### 3. Test Separator and Quote “Cell Breakout” Scenarios

Because CSV is cell-based, test whether you can inject content that starts a new cell and then begins with a dangerous character. This depends on:

- Field separator (commonly `,` or `;`)
- Quoting rules and escaping
- Application-side CSV generation and encoding

Example *benign* test patterns (adjust separator to the actual export format):

- A value containing a quote and separator intended to create a new cell, then `=1+1`
- A value containing a separator directly (if not quoted by the exporter), then `=1+1`

Your objective is to see whether the resulting CSV contains any cell whose first character is one of the formula-triggering prefixes. Verify this by inspecting the raw CSV output in a text editor.

### 4. Export and Verify in Spreadsheet Applications

- Export/download the CSV.
- Open it in at least one spreadsheet application used in the target environment (Excel and/or LibreOffice, etc.).
- Confirm whether the cell is interpreted as a formula:
    - The spreadsheet displays the computed result (e.g., `2`) instead of the literal string (e.g., `=1+1`), and/or
    - The formula bar shows a formula rather than plain text.

Record:

- Application name/version (e.g., Microsoft Excel)
- How the file was opened (double-click vs import wizard)
- Whether locale settings influenced parsing

### 5. Excel Save/Re-open Regression Test (Mitigation Reliability)

If the application claims to escape/quote values:

- Open the exported CSV in Microsoft Excel.
- Save the file (e.g., as CSV).
- Close and re-open it.
- Re-check whether previously “neutralized” values became active formulas.

This step is critical because Excel may normalize/remove certain escaping/quoting behaviors after save/reopen.

### 6. Assess Impact

Assess practical risk based on:

- Who opens the file (privileged roles vs end users)
- Whether exports are shared externally
- Whether spreadsheet values are relied on for decisions (e.g., totals, flags, statuses)
- Whether formulas could mislead users or trigger spreadsheet behaviors in your environment

Provide a clear reproduction path:

1) Where input is entered
2) How export is generated
3) Which program interprets it as a formula
4) Evidence (screenshot or description) that evaluation occurs

### 7. Higher-Impact Validation (Optional, Strictly Controlled)

Some real-world cases show formula injection escalating beyond simple calculation/deception (e.g., via legacy spreadsheet features such as DDE). This behavior is highly environment-dependent.

- Do **not** embed command-execution formulas in this guide.
- If demonstrating higher impact is explicitly authorized, perform it only in an isolated test environment and cite public research/bug-bounty reports rather than reproducing weaponizable payloads in documentation.

## Remediation

There is no universal CSV sanitization strategy safe for all spreadsheet applications and downstream consumers. For CSVs intended for human viewing in spreadsheet software, apply a defense-in-depth approach:

- Ensure no cell begins with formula-triggering characters (`=`, `+`, `-`, `@`) and relevant control/Unicode variants.
- Consider per-field sanitization (commonly suggested):
    - Wrap each cell in double quotes
    - Prepend each cell with a single quote
    - Escape double quotes by doubling them  
  Note: This may not remain reliable in Microsoft Excel after saving and re-opening.

- Excel-resistant mitigation (observed behavior):
    - If a cell starts with `=`, `+`, `-`, or `@`, prefix the content with a tab character (`0x09`) inside a quoted field (e.g., `"\t=1+1"`).  
  Trade-off: The tab becomes part of the underlying data and may affect downstream programmatic imports.

Always validate the chosen mitigation against the spreadsheet applications and workflows in scope.

## Tools

- Intercepting proxy (e.g., Burp Suite, OWASP ZAP) to inject controlled values and reproduce the exact export workflow.
- Spreadsheet software for validation in an isolated environment (e.g., Microsoft Excel, LibreOffice Calc).
- A safe HTTP listener (local or internal test host) to observe clicks/requests when validating `HYPERLINK()` behavior.
- A text editor to inspect the raw CSV and confirm whether any cell begins with a formula-triggering prefix.

## References

- [OWASP CSV Injection](https://owasp.org/www-community/attacks/CSV_Injection)
- [CWE-1236: Improper Neutralization of Formula Elements in a CSV File](https://cwe.mitre.org/data/definitions/1236.html)
- [HackerOne Report #72785: CSV Injection](https://hackerone.com/reports/72785)
- [HackerOne Report #118582: CSV Injection](https://hackerone.com/reports/118582)
- [HackerOne Report #1748961: CSV Injection](https://hackerone.com/reports/1748961)
- [HackerOne Report #244292: CSV Injection](https://hackerone.com/reports/244292)
```

**Comment by kingthorin** [comment]
Agreed, you can add commits addressing stuff as needed.

**Comment by m4sh-wacker** [comment]
Hi @Galaxy-sc, thank you for help! I've applied the suggestions.

**Comment by m4sh-wacker** [comment]
Hi @kingthorin,
Just checking in to see if any further changes or inputs are required from my side for this PR.
I understand reviewing a new section takes time, but I'm available to address any feedback immediately.

Thanks

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
+# Testing for CSV Injection
+
+|ID|
+|--|
+|WSTG-INPV-21|
+
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]: This should match other occurrences in the guide.

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+## Summary
+
+CSV Injection (also known as Formula Injection) occurs when an application embeds untrusted, user-controlled input into CSV (or other spreadsheet-compatible) exports and the resulting file is opened in a spreadsheet program (e.g., Microsoft Excel, LibreOffice Calc). Spreadsheet applications may interpret certain cell values as formulas, which can lead to security issues such as user deception (phishing-style workflows), manipulation of spreadsheet output, or data exfiltration. In some environments, formula injection can be escalated to higher impact via spreadsheet “gadgets” and legacy features (e.g., DDE / Dynamic Data Exchange behaviors), potentially reaching command execution on the workstation that opens the file—typically dependent on client configuration and/or user interaction.
+
+A key characteristic of this issue is that the vulnerability often manifests only when the exported file is opened by a user (e.g., an administrator, finance, or support) in a spreadsheet application.
+
+### Formula-triggering prefixes
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]: This should use proper title caps

Comment (kingthorin) [reply document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]: This whole section should be relocated so that the scenario follows the template flow.

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+- Identify CSV/spreadsheet export features that include untrusted input.
+- Verify whether attacker-controlled values are interpreted as formulas when the export is opened in common spreadsheet applications.
+- Check whether separator/quote injection can move a dangerous prefix to the start of a cell.
+- Validate whether mitigations remain effective in Microsoft Excel after saving and re-opening the CSV.
+- Assess practical impact based on who opens the export and how it is used.
+
+## How to Test
+
+### 1. Identify CSV Export Functionality and Data Sources
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]: The numbers should be dropped from these headings.

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+- Whether exports are shared externally
+- Whether spreadsheet values are relied on for decisions (e.g., totals, flags, statuses)
+- Whether formulas could mislead users or trigger spreadsheet behaviors in your environment
+
+Provide a clear reproduction path:
+
+1) Where input is entered
+2) How export is generated
+3) Which program interprets it as a formula
+4) Evidence (screenshot or description) that evaluation occurs
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]: This should use the number period format for ordered list.

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+### 7. Higher-Impact Validation (Optional, Strictly Controlled)
+
+Some real-world cases show formula injection escalating beyond simple calculation/deception (e.g., via legacy spreadsheet features such as DDE). This behavior is highly environment-dependent.
+
+- Do **not** embed command-execution formulas in this guide.
+- If demonstrating higher impact is explicitly authorized, perform it only in an isolated test environment and cite public research/bug-bounty reports rather than reproducing weaponizable payloads in documentation.
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]: This can be removed.

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+- Excel-resistant mitigation (observed behavior):
+    - If a cell starts with `=`, `+`, `-`, or `@`, prefix the content with a tab character (`0x09`) inside a quoted field (e.g., `"\t=1+1"`).  
+  Trade-off: The tab becomes part of the underlying data and may affect downstream programmatic imports.
+
+Always validate the chosen mitigation against the spreadsheet applications and workflows in scope.
+
+## Tools
+
+- Intercepting proxy (e.g., Burp Suite, OWASP ZAP) to inject controlled values and reproduce the exact export workflow.
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]: ZAP left OWASP years ago.

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/README.md [hunk 8]
```diff
@@ -57,3 +57,5 @@
 4.7.19 [Testing for Server-Side Request Forgery](19-Testing_for_Server-Side_Request_Forgery.md)
 
 4.7.20 [Testing for Mass Assignment](20-Testing_for_Mass_Assignment.md)
+
+4.7.21 [Testing for CSV Injection](21-Testing_for_CSV_Injection.md)
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/README.md:61]: Should also be added to the main index.

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 9]
```diff
@@ -0,0 +1,172 @@
+# Testing for CSV Injection
+
+|ID|
+|--|
+|WSTG-INPV-21|
```
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]:
```suggestion
|ID          |
|------------|
|WSTG-INPV-21|
```

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md (hunk 4 above)
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]:
```suggestion
### Identify CSV Export Functionality and Data Sources
```

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+Identify untrusted data sources that can end up in the export:
+
+- User profiles (name, email, company)
+- Free-text fields (comments, ticket subjects, notes)
+- Imported/integrated external data (webhooks, CRM sync, partner feeds)
+
+Document which roles can trigger the export and which roles are likely to open it.
+
+### 2. Place Benign, Detectable Formula-Like Values into Candidate Fields
```
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]:
```suggestion
### Place Benign, Detectable Formula-Like Values into Candidate Fields
```

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+
+- The `HYPERLINK()` case is useful to demonstrate realistic impact (deception/phishing-style flows, or potential metadata exposure when a link is clicked/opened). Use a controlled endpoint during testing (e.g., a local listener or an internal test host).
+- Do not use external “real attacker” infrastructure in validation.
+
+Also test control-character and Unicode variants (where input handling allows it):
+
+- A value that begins with a tab character followed by `=1+1` (TAB + `=1+1`)
+- Full-width prefix variants (e.g., `＝1+1`)
+
+### 3. Test Separator and Quote “Cell Breakout” Scenarios
```
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]:
```suggestion
### Test Separator and Quote “Cell Breakout” Scenarios
```

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+- Application-side CSV generation and encoding
+
+Example *benign* test patterns (adjust separator to the actual export format):
+
+- A value containing a quote and separator intended to create a new cell, then `=1+1`
+- A value containing a separator directly (if not quoted by the exporter), then `=1+1`
+
+Your objective is to see whether the resulting CSV contains any cell whose first character is one of the formula-triggering prefixes. Verify this by inspecting the raw CSV output in a text editor.
+
+### 4. Export and Verify in Spreadsheet Applications
```
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]:
```suggestion
### Export and Verify in Spreadsheet Applications
```

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+    - The spreadsheet displays the computed result (e.g., `2`) instead of the literal string (e.g., `=1+1`), and/or
+    - The formula bar shows a formula rather than plain text.
+
+Record:
+
+- Application name/version (e.g., Microsoft Excel)
+- How the file was opened (double-click vs import wizard)
+- Whether locale settings influenced parsing
+
+### 5. Excel Save/Re-open Regression Test (Mitigation Reliability)
```
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]:
```suggestion
### Excel Save/Re-open Regression Test (Mitigation Reliability)
```

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+If the application claims to escape/quote values:
+
+- Open the exported CSV in Microsoft Excel.
+- Save the file (e.g., as CSV).
+- Close and re-open it.
+- Re-check whether previously “neutralized” values became active formulas.
+
+This step is critical because Excel may normalize/remove certain escaping/quoting behaviors after save/reopen.
+
+### 6. Assess Impact
```
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]:
```suggestion
### Assess Impact
```

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+- Whether formulas could mislead users or trigger spreadsheet behaviors in your environment
+
+Provide a clear reproduction path:
+
+1) Where input is entered
+2) How export is generated
+3) Which program interprets it as a formula
+4) Evidence (screenshot or description) that evaluation occurs
+
+### 7. Higher-Impact Validation (Optional, Strictly Controlled)
```
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]:
```suggestion
### Higher-Impact Validation (Optional, Strictly Controlled)
```

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md (hunk 7 above)
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]:
```suggestion
- Intercepting proxy (e.g., Burp Suite, ZAP) to inject controlled values and reproduce the exact export workflow.
```

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md (hunk 5 above)
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]:
```suggestion
1. Where input is entered
2. How export is generated
3. Which program interprets it as a formula
4. Evidence (screenshot or description) that evaluation occurs
```

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 16]
```diff
@@ -0,0 +1,172 @@
...
+4. Evidence (screenshot or description) that evaluation occurs
+
+### Higher-Impact Validation (Optional, Strictly Controlled)
+
+Some real-world cases show formula injection escalating beyond simple calculation/deception (e.g., via legacy spreadsheet features such as DDE). This behavior is highly environment-dependent.
+
+- Do **not** embed command-execution formulas in this guide.
+- If demonstrating higher impact is explicitly authorized, perform it only in an isolated test environment and cite public research/bug-bounty reports rather than reproducing weaponizable payloads in documentation.
+
+## Remediation
```
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md:138]:
```suggestion
### Higher-Impact Validation (Optional, Strictly Controlled)

Some real-world cases show formula injection escalating beyond simple calculation/deception (e.g., via legacy spreadsheet features such as DDE). This behavior is highly environment-dependent.

## Remediation
```

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 17]
```diff
@@ -0,0 +1,169 @@
...
+- Tab (`0x09`)
+- Carriage return (`0x0D`)
+- Line feed (`0x0A`)
+- Full-width (double-byte) variants such as `＝`, `＋`, `－`, `＠` (depending on locale/application behavior)
+
+> Important (Excel behavior): Microsoft Excel may remove quotes or escape characters from CSV cells when a file is saved and re-opened. As a result, some commonly suggested mitigations can fail after save/reopen and previously escaped formulas may become active again.
+
+Also note that it is not sufficient to ensure the *overall* untrusted input does not start with a dangerous character. Attackers may inject separators and quoting to start a new cell, placing the dangerous character at the beginning of a cell.
+
+## Test Objectives
```
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md]:
```suggestion
## Test Objectives
```

File: document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md [hunk 18]
```diff
@@ -0,0 +1,169 @@
...
+
+- Identify CSV/spreadsheet export features that include untrusted input.
+- Verify whether attacker-controlled values are interpreted as formulas when the export is opened in common spreadsheet applications.
+- Check whether separator/quote injection can move a dangerous prefix to the start of a cell.
+- Validate whether mitigations remain effective in Microsoft Excel after saving and re-opening the CSV.
+- Assess practical impact based on who opens the export and how it is used.
+
+## How to Test
+
+### Identify CSV Export Functionality and Data Sources
```
Comment (Galaxy-sc) [inline document/4-Web_Application_Security_Testing/07-Input_Validation_Testing/21-Testing_for_CSV_Injection.md:40]:
```suggestion
## How to Test

### Formula-Triggering Prefixes

Cells beginning with the following characters may be interpreted as formulas by spreadsheet software:

- Equals (`=`)
- Plus (`+`)
- Minus (`-`)
- At (`@`)
- Tab (`0x09`)
- Carriage return (`0x0D`)
- Line feed (`0x0A`)
- Full-width (double-byte) variants such as `＝`, `＋`, `－`, `＠` (depending on locale/application behavior)

> Important (Excel behavior): Microsoft Excel may remove quotes or escape characters from CSV cells when a file is saved and re-opened. As a result, some commonly suggested mitigations can fail after save/reopen and previously escaped formulas may become active again.

Also note that it is not sufficient to ensure the *overall* untrusted input does not start with a dangerous character. Attackers may inject separators and quoting to start a new cell, placing the dangerous character at the beginning of a cell.

### Identify CSV Export Functionality and Data Sources
```

**Comment by Galaxy-sc** [comment]
@kingthorin All the requested changes (Title caps, relocating the prefixes section, updating ZAP, and fixing the main indexes) have been applied. The branch is also synced with the upstream master to prevent conflicts. 

It should be ready for another look. Thanks again for the review!

**Comment by m4sh-wacker** [comment]
Hi @Galaxy-sc 
Thanks for the quick fix — much appreciated!

**Comment by m4sh-wacker** [comment]
Hi @kingthorin All issues are fixed and the updates are in place.
Ready for re-review whenever you are.

**Comment by kingthorin** [comment]
Sorry for the rebase spam, I'm trying to get something fixed.

**Comment by kingthorin** [comment]
@m4sh-wacker I think I got the workflow issues sorted, thanks for putting up with all the rebases. I'll try to review this tonight or early tomorrow.

**Comment by m4sh-wacker** [comment]
@kingthorin Thanks, I appreciate it.
Happy to make any changes if needed after your review.
